shared-module/displayio/Bitmap.c

```diff
@@ -33,20 +33,21 @@
 void common_hal_displayio_bitmap_construct(displayio_bitmap_t *self, uint32_t width,
     uint32_t height, uint32_t bits_per_value) {
     uint32_t row_width = width * bits_per_value;
-    // word align
-    if (row_width % 32 != 0) {
-        self->stride = (row_width / 32 + 1);
+    // align to size_t
+    uint8_t align_bits = 8 * sizeof(size_t);
+    if (row_width % align_bits != 0) {
+        self->stride = (row_width / align_bits + 1);
     } else {
-        self->stride = row_width / 32;
+        self->stride = row_width / align_bits;
     }
     self->width = width;
     self->height = height;
-    self->data = m_malloc(self->stride * height * sizeof(uint32_t), false);
+    self->data = m_malloc(self->stride * height * sizeof(size_t), false);
     self->read_only = false;
     self->bits_per_value = bits_per_value;
 
-    if (bits_per_value > 8) {
-        mp_raise_NotImplementedError(translate("Only bit maps of 8 bit color or less are supported"));
+    if (bits_per_value > 8 && bits_per_value != 16 && bits_per_value != 32) {
```

Comment:
Doesn't this assume that ``size_t`` is ``uint32_t``?

Reply:
Also, shouldn't we also be checking that it's a power of two when < 8?

Reply:
How does it assume uint32_t is size_t?

It's probably better to modify this check to check for 24 bit and fail. We can rely on shared-bindings to do power of two below 8 and multiple of 8 above. I just didn't want to support packing/unpacking 24-bit values.

Reply:
Well, it actually assumes ``32 <= size_t <= 40``. I guess I'm trying to understand why you replaced ``uint32_t`` with ``size_t`` earlier on, when you still hard-code the supported sizes.

Reply:
I changed it because I was in there and better understand size_t can be used to support 64 bit in the future. Perhaps I should have left it to reduce the diff size.

Reply:
I see, sorry, I thought it's related.